### PR TITLE
Run code sniffer against future v3

### PR DESCRIPTION
**Pull request by antograssiot**
Worked on my fork so hopefully will work here also

### CODE REVIEW

**Comment by ADmad** [comment]
The job using codesniffer v3 seems to be running now and it found errors https://travis-ci.org/cakephp/cakephp/jobs/189078059#L642  :smile:

File: .travis.yml [hunk 1]
```diff
@@ -87,7 +87,8 @@ before_install:
   - set +H
 
 before_script:
-  - if [[ $PHPCS != 3 ]]; then composer install --prefer-dist --no-interaction; fi
+  - composer install --prefer-dist --no-interaction
+  - if [[ $PHPCS = 3 ]]; then composer remove --dev cakephp/cakephp-codesniffer; fi
   - if [[ $PHPCS = 3 ]]; then composer require --dev  squizlabs/php_codesniffer=3.0.x-dev@dev cakephp/cakephp-codesniffer=3.0.x-dev@dev; fi
```
Comment (ADmad) [inline .travis.yml]: Instead of `3.0.x-dev@dev` you can use `^3.0.*@dev` which is a bit shorter :)

**Comment by antograssiot** [comment]
Yeah those two errorseparate were expected. 
Now we have two ways to fix them I guess:
 - @ignore.. annotation 
 - split the public method name sniff  (one for classes and  one for traits) and use the phpcs.xml

The first looks better to me, WDYT ?

File: .travis.yml (hunk 1 above)
Comment (antograssiot) [inline .travis.yml]: Thx will update it

**Comment by ADmad** [comment]
Adding annotation to ignore sounds good to me.

**Comment by antograssiot** [comment]
PR updated, the build should be passing now.

File: .travis.yml [hunk 2]
```diff
@@ -87,8 +87,9 @@ before_install:
   - set +H
 
 before_script:
-  - if [[ $PHPCS != 3 ]]; then composer install --prefer-dist --no-interaction; fi
-  - if [[ $PHPCS = 3 ]]; then composer require --dev  squizlabs/php_codesniffer=3.0.x-dev@dev cakephp/cakephp-codesniffer=3.0.x-dev@dev; fi
+  - composer install --prefer-dist --no-interaction
+  - if [[ $PHPCS = 3 ]]; then composer remove --dev cakephp/cakephp-codesniffer; fi
+  - if [[ $PHPCS = 3 ]]; then composer require --dev squizlabs/php_codesniffer='3.0.*@dev' cakephp/cakephp-codesniffer='3.0.*@dev'; fi
```
Comment (dereuromark) [inline .travis.yml]: Don't we want `^3.0.0@dev` ?

Comment (antograssiot) [inline .travis.yml]: We want stable :P

Comment (dereuromark) [inline .travis.yml]: Hehe, that too :P

**Pull request merged**